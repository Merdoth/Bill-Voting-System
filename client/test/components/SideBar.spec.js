import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { SideBar } from
  '../../js/components/SideBar';

const logout = jest.fn();
const event = {
  preventDefault: jest.fn(),
};
const setup = permission => shallow(<SideBar logout={logout} permission={permission} />);

describe('SideBar Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup(3);
    expect(wrapper).toMatchSnapshot();
  });

  describe('logout()', () => {
    it('should logout a user', () => {
      [1, 2, 3].forEach((permission) => {
        const wrapper = setup(permission);
        const button = wrapper.find('#logout');
        button.simulate('click', event);
        expect(logout).toHaveBeenCalled();
      });
    });
  });
});
