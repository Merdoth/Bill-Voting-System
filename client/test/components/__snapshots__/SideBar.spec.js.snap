// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component snapshot it should render the right amount of elements 1`] = `
<div>
  <header>
    <div
      className="side-nav fixed"
      id="slide-out"
    >
      <div
        className="side-nav-section logo"
      >
        <Link
          className="brand-logo logo"
          replace={false}
          to="/"
        >
          <img
            alt="test"
            height="40"
            src="/img/logo-billvoting.png"
          />
          <span
            className="logo-text"
          >
            wevote
          </span>
        </Link>
      </div>
      <div
        className="side-nav-section channels"
      >
        <ul
          className="side-nav-list"
        >
          <li
            className="side-nav-item"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              id="user-profile"
              to="/profile"
            >
              Profile 
            </NavLink>
          </li>
          <li
            className="side-nav-item"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/bills"
            >
              Bills 
            </NavLink>
          </li>
          <li
            className="side-nav-item"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              id="user-votedBills"
              to="/bills/voted"
            >
              Voted Bills 
            </NavLink>
          </li>
        </ul>
        <div
          className="side-nav-logout-btn"
          id="logout"
          onClick={[Function]}
        >
          <a>
            Logout
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
